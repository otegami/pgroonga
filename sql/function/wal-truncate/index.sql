SET pgroonga.enable_wal = yes;

CREATE TABLE memos (
  content text
);

INSERT INTO memos VALUES ('Groonga is fast!');

CREATE INDEX pgrn_index ON memos USING PGroonga (content);

INSERT INTO memos VALUES ('PGroonga is also fast!');

SELECT pgroonga_command('truncate',
                        ARRAY[
                          'table', pgroonga_table_name('pgrn_index')
                        ])::jsonb->>1;

SELECT pgroonga_wal_truncate('pgrn_index');

INSERT INTO memos VALUES ('PostgreSQL is a good RDBMS!');

SELECT pgroonga_command('delete',
                        ARRAY[
                          'table', 'IndexStatuses',
                          'key', 'pgrn_index'::regclass::oid::text
                        ])::jsonb->>1;
SELECT pgroonga_command('truncate',
                        ARRAY[
                          'table', pgroonga_table_name('pgrn_index')
                        ])::jsonb->>1;

SELECT pgroonga_wal_apply('pgrn_index');

SELECT pgroonga_command('select',
                        ARRAY[
                          'table', pgroonga_table_name('pgrn_index'),
                          'output_columns', '_id, content'
                        ])::jsonb->>1;

DROP TABLE memos;

SET pgroonga.enable_wal = default;
