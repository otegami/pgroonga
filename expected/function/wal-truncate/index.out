SET pgroonga.enable_wal = yes;
CREATE TABLE memos (
  content text
);
INSERT INTO memos VALUES ('Groonga is fast!');
CREATE INDEX pgrn_index ON memos USING PGroonga (content);
INSERT INTO memos VALUES ('PGroonga is also fast!');
SELECT pgroonga_command('truncate',
                        ARRAY[
                          'table', pgroonga_table_name('pgrn_index')
                        ])::jsonb->>1;
 ?column? 
----------
 true
(1 row)

SELECT pgroonga_wal_truncate('pgrn_index');
 pgroonga_wal_truncate 
-----------------------
                     2
(1 row)

INSERT INTO memos VALUES ('PostgreSQL is a good RDBMS!');
SELECT pgroonga_command('delete',
                        ARRAY[
                          'table', 'IndexStatuses',
                          'key', 'pgrn_index'::regclass::oid::text
                        ])::jsonb->>1;
 ?column? 
----------
 true
(1 row)

SELECT pgroonga_command('truncate',
                        ARRAY[
                          'table', pgroonga_table_name('pgrn_index')
                        ])::jsonb->>1;
 ?column? 
----------
 true
(1 row)

SELECT pgroonga_wal_apply('pgrn_index');
 pgroonga_wal_apply 
--------------------
                  1
(1 row)

SELECT pgroonga_command('select',
                        ARRAY[
                          'table', pgroonga_table_name('pgrn_index'),
                          'output_columns', '_id, content'
                        ])::jsonb->>1;
                                         ?column?                                          
-------------------------------------------------------------------------------------------
 [[[1], [["_id", "UInt32"], ["content", "LongText"]], [1, "PostgreSQL is a good RDBMS!"]]]
(1 row)

DROP TABLE memos;
SET pgroonga.enable_wal = default;
